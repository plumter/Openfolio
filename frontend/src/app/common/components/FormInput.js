import InputError from "app/common/components/InputError";
import InputWithFix from "app/common/components/InputWithFix";

const FormInput = ({label, invalidMessage, validation, ...props}) => {

    return  <div>
                {
                    label &&    <label className="form-label">
                                    {label}
                                </label>
                }
                <InputWithFix
                    type='text' 
                    {...props}
                />
                <InputError
                    name={props.name}
                    validation={validation}
                    text={invalidMessage ?? props.placeholder}
                />
            </div>
}

export default FormInput;