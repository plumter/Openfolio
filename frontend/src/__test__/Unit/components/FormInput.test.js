import { create } from 'react-test-renderer';
import SVG from "react-inlinesvg";
import FormInput from 'app/common/components/FormInput';

describe("FormInput Component Unit Testing", _ => {

    it('Test FormInput Cases',  () => {
      // 1st case
      const component = create(      
          <FormInput
            label="Address:"
            placeholder="Enter Company Address" 
            name="address"
            minLength="2"
            key={1}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 2nd case
      component.update(      
          <FormInput
            label={<>Company Name:<span className="text-danger">*</span></>}
            placeholder="Enter Company Name" 
            name="company"
            required
            minLength="2"
            validation={{}}
            prefix={<span className="border-r pr-2 pl-1 pb-1 inline-block dark:border-r-dark"> 
                    <SVG
                        src="/assets/media/svg/company.svg"
                        className="stroke-current w-5 inline-block"
                    />
                </span>}
            key={2}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 3rd case
      component.update(      
          <FormInput
            type="url"
            label="Website:"
            placeholder="Enter website url" 
            name="website"
            validation={{}}
            invalidMessage="Enter valid Url"
              key={3}
          />,
        );

        // 4th case
        component.update(      
            <FormInput
                label={<>Name:<span className="text-danger">*</span></>}
                placeholder="Enter your Name" 
                name="name"
                required
                minLength="2"
                validation={{}}
                key={4}
            />,
          );

          // 5th case
          component.update(      
              <FormInput
                label={<>Email:<span className="text-danger">*</span></>}
                type='email' 
                placeholder="example@mail.com" 
                required
                name="email"
                value="chineduegenti@gmail.com"
                readOnly
                validation={{}}
                key={5}
              />,
            );
      expect(component.toJSON()).toMatchSnapshot();
    });  
});