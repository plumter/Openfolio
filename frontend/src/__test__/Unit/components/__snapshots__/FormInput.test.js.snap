// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component Unit Testing Test FormInput Cases 1`] = `
<div>
  <label
    className="form-label"
  >
    Address:
  </label>
  <div
    className="relative my-auto"
  >
    <input
      className="form-input undefined undefined"
      minLength="2"
      name="address"
      placeholder="Enter Company Address"
      type="text"
    />
  </div>
  <div
    style={
      Object {
        "height": 0,
        "visibility": "hidden",
      }
    }
  >
    <div
      className="react-reveal text-xs mb-1 pt-2 text-red-500"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      Enter Company Address
    </div>
  </div>
</div>
`;

exports[`FormInput Component Unit Testing Test FormInput Cases 2`] = `
<div>
  <label
    className="form-label"
  >
    Company Name:
    <span
      className="text-danger"
    >
      *
    </span>
  </label>
  <div
    className="relative my-auto"
  >
    <div
      className="absolute inset-y-0 left-0 flex items-center px-2"
    >
      <label
        className=" rounded font-medium px-2 pt-2.5 text-sm dark:text-white cursor-pointer z-30"
      >
        <span
          className="border-r pr-2 pl-1 pb-1 inline-block dark:border-r-dark"
        />
      </label>
    </div>
    <input
      className="form-input pl-14 undefined"
      minLength="2"
      name="company"
      placeholder="Enter Company Name"
      required={true}
      type="text"
    />
  </div>
  <div
    style={
      Object {
        "height": 0,
        "visibility": "hidden",
      }
    }
  >
    <div
      className="react-reveal text-xs mb-1 pt-2 text-red-500"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      Enter Company Name
    </div>
  </div>
</div>
`;

exports[`FormInput Component Unit Testing Test FormInput Cases 3`] = `
<div>
  <label
    className="form-label"
  >
    Email:
    <span
      className="text-danger"
    >
      *
    </span>
  </label>
  <div
    className="relative my-auto"
  >
    <input
      className="form-input undefined undefined"
      name="email"
      placeholder="example@mail.com"
      readOnly={true}
      required={true}
      type="email"
      value="chineduegenti@gmail.com"
    />
  </div>
  <div
    style={
      Object {
        "height": 0,
        "visibility": "hidden",
      }
    }
  >
    <div
      className="react-reveal text-xs mb-1 pt-2 text-red-500"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      example@mail.com
    </div>
  </div>
</div>
`;
